### DISCUSSION TITLE
Make YES default button when file is deleting

### DISCUSSION THREAD

**Discussion by ghost** [lxqt/lxqt#2103] [Q&A]
Currently NO is selected by default:

![image](https://user-images.githubusercontent.com/4306334/138983119-604ca2e0-7560-47c7-9fb3-3ed482198dd5.png)

How to set YES in options?

**Comment by stefonarch** [comment]
IMO this is a security feature and shouldn't be changed. One accidental hit on "enter" and the files would be gone.
To change this an option would be an overkill. Only way would be changing it in the code.

**Reply by tsujan** [reply]
I completely agree with @stefonarch.

Pressing a "Y" lets the user make sure about deleting his/her files — which can be valuable to him — without wasting his time. In contrast, `Enter` or `Return` is too neutral.

**Reply by ghost** [reply]
What we have for the moment is _double confirmation_.  
If you accidentally press delete then most likely you will not accidentally press Enter.  
But in case of default No button, you have once again say "yes delete that file".  
You have option to not ask for confirmation at all.  
About which security we are talking about?  ))

Y keyboard button not OK because it is small. And located not close to Delete and other navigation keys.  
It makes sense when you need to organize directories and move and delete files and so on.  

So I need to replay default Windows behaviour.  
I am sure it must be configurable to get less actions to achieve the same result.

To avoid any troubles with existing users default behaviour can use No as default.

**Reply by tsujan** [reply]
Please don't aggrandize this as if it's torturous. You don't like it; someone else doesn't like the opposite. We don't add options for such things.

**Reply by stefonarch** [reply]
> What we have for the moment is double confirmation.

Do you talk about delete or trash? These are actually the options in Preferences, I didn't notice that it's possible to avoid any confirmation for erasing.

![schermata-10-27-14-39](https://user-images.githubusercontent.com/10681413/139067538-ef77e8f2-f0c5-475a-8673-b7c8327bed60.png)

**Reply by tsujan** [reply]
>  I didn't notice that it's possible to avoid any confirmation for erasing

It's an old option for those that feel infallible. I didn't remove it because it was there before I came to LXQt (the philosophy of respecting the original author's choices).

**Comment by ghost** [comment]
> Do you talk about delete or trash? 

Trash is disabled.

I just told about the need to perform 2 actions to delete some file:

1) Focus on YES button.
2) Press Enter to finally confirm.

It is the same as you show 2 popups (step by step):

1) Are sure you want to delete the file?
2) Are you really want to delete it?

99.99% of deletion requests from a user should result in deletion of a file.

So when confirmation is required it should not be annoying. I should not perform 2 actions to confirm my initial intention.

Hopefully it is clear now.

**Reply by tsujan** [reply]
> I just told about the need to perform 2 actions to delete some file

As iI's mentioned above that it isn't so: press "Y".

**Reply by ghost** [reply]
But not Enter as it is everywhere.

How do you think it is easy to do on small notebooks?

**Comment by ghost** [comment]
> Please don't aggrandize this as if it's torturous. You don't like it; someone else doesn't like the opposite. We don't add options for such things.

Please don't underestimate this as if usability is nothing. You do not like new options for user; someone else doesn't like the opposite.   
Probably we don't need to grow community and user-base. Though in that case, I will hand over my tickets for the Titanic.

**Reply by tsujan** [reply]
Those who want options for everything may use KDE. Those who don't want options for anything may use Gnome. LXQt adds options but not for everything.

P.S.
This is a Discussion page that may be helpful to those who find it by searching in Internet. Repeating comments aren't useful in it.
